code/DDSCodeTester.cpp

```diff
@@ -1156,6 +1158,29 @@ class RemoteDiscoveryDomainParticipantListener : public DomainParticipantListene
 };
 //!--
 
+//!--REMOTE_TYPE_INTROSPECTION
+class TypeIntrospectionSubscriber : public DomainParticipantListener
+{
+    /* Custom Callback on_data_available */
+    void on_data_available(
+            DataReader* reader)
+    {
+        // Dynamic DataType
+        DynamicData::_ref_type new_data =
+                            DynamicDataFactory::get_instance()->create_data(dyn_type_);
+
+        std::stringstream output;
+        output << std::setw(4);
+
+        // Serialize DynamicData into JSON string format
+        json_serialize(new_data, DynamicDataJsonFormat::EPROSIMA, output);
+        std::cout << "Message received:\n" << output.str() << std::endl;
+    }
```

Comment:
I'd show only this code in the documentation (do not forget to remove extra indents) as the example is about JSON serialization and not about Type Introspection.